sdk/state/state_test.go

```diff
@@ -391,6 +414,177 @@ func Test(t *testing.T) {
+		t.Log("Proposal: ", i, paymentLog, amount/1_000_0000)
+
+		// Sender: creates new Payment, sends to other party
+		payment, err := sendingChannel.ProposePayment(state.Amount{Asset: state.NativeAsset{}, Amount: amount})
+		require.NoError(t, err)
+
+		var fullySigned bool
+
+		// Receiver: receives new payment, validates, then confirms by signing both
+		payment, fullySigned, err = receivingChannel.ConfirmPayment(payment)
+		require.NoError(t, err)
+		require.False(t, fullySigned)
+
+		// Sender: re-confirms P_i by signing D_i and sending back
+		payment, fullySigned, err = sendingChannel.ConfirmPayment(payment)
+		require.NoError(t, err)
+		require.True(t, fullySigned)
+
+		// Receiver: receives new payment, validates, then confirms by signing both
+		payment, fullySigned, err = receivingChannel.ConfirmPayment(payment)
+		require.NoError(t, err)
+		require.True(t, fullySigned)
+		ci, di, err := sendingChannel.PaymentTxs(payment)
+		require.NoError(t, err)
+		ci, err = ci.AddSignatureDecorated(payment.CloseSignatures...)
+		require.NoError(t, err)
+		di, err = di.AddSignatureDecorated(payment.DeclarationSignatures...)
+		require.NoError(t, err)
+	}
+
+	// Coordinated Close
+	t.Log("Begin coordinated close process ...")
+	t.Log("Initiator submitting latest declaration transaction")
+	lastD, _, err := initiatorChannel.CloseTxs()
+	require.NoError(t, err)
+	lastD, err = lastD.AddSignatureDecorated(initiatorChannel.LatestCloseAgreement().DeclarationSignatures...)
+	require.NoError(t, err)
+
+	fbtx, err := txnbuild.NewFeeBumpTransaction(txnbuild.FeeBumpTransactionParams{
+		Inner:      lastD,
+		FeeAccount: initiator.KP.Address(),
+		BaseFee:    txnbuild.MinBaseFee,
+	})
+	require.NoError(t, err)
+	fbtx, err = fbtx.Sign(networkPassphrase, initiator.KP)
+	require.NoError(t, err)
+	_, err = client.SubmitFeeBumpTransaction(fbtx)
+	require.NoError(t, err)
+
+	t.Log("Initiator proposes a coordinated close")
+	cc, err := initiatorChannel.ProposeCoordinatedClose(0, 0)
+	require.NoError(t, err)
+	cc, err = responderChannel.ConfirmCoordinatedClose(cc)
+	require.NoError(t, err)
+	cc, err = initiatorChannel.ConfirmCoordinatedClose(cc)
+	require.NoError(t, err)
+
+	t.Log("Initiator closing channel with new coordinated close transaction")
+	txCoordinated, err := initiatorChannel.CoordinatedCloseTx()
+	require.NoError(t, err)
+	txCoordinated, err = txCoordinated.AddSignatureDecorated(initiatorChannel.CoordinatedClose().CloseSignatures()...)
+	fbtx, err = txnbuild.NewFeeBumpTransaction(txnbuild.FeeBumpTransactionParams{
+		Inner:      txCoordinated,
+		FeeAccount: initiator.KP.Address(),
+		BaseFee:    txnbuild.MinBaseFee,
+	})
+	require.NoError(t, err)
+	fbtx, err = fbtx.Sign(networkPassphrase, initiator.KP)
+	require.NoError(t, err)
+	_, err = client.SubmitFeeBumpTransaction(fbtx)
+	if err != nil {
+		hErr := horizonclient.GetError(err)
+		t.Log("Submitting Close:", txCoordinated.SourceAccount().Sequence, "Error:", err)
+		t.Log(hErr.ResultString())
```

Comment:
do we need to return if this errors out, or is it meaningful to continue through the end of this test?

Reply:
oops good catch, we should end it
https://github.com/stellar/experimental-payment-channels/pull/83/commits/5fa3ee4c71d32837169639010cf138ec38b71988